### PR TITLE
Fix `explain` panic for non-existent collection on PostgreSQL

### LINKED ISSUE #3415: Implement sort pushdown
### What should be done?

We should implement collection query sort pushdown as in `pgdb`.

Relevant integration tests should pass.

### Where?

https://github.com/FerretDB/FerretDB/blob/main/internal/backends/postgresql/collection.go

### Definition of Done

- unit tests added/updated;
- integration/compatibility tests added/updated;
- spot refactorings done.

### PR DESCRIPTION
**Pull request by noisersup**
# Description

Closes #3415.

## Readiness checklist

- [ ] I added/updated unit tests (and they pass).
- [x] I added/updated integration/compatibility tests (and they pass).
- [ ] I added/updated comments and checked rendering.
- [x] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Milestone (`Next`), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

**Review by AlekSi — COMMENTED** [review]
conform-pr fails

File: internal/backends/postgresql/collection.go [hunk 1]
```diff
@@ -261,13 +261,15 @@ func (c *collection) DeleteAll(ctx context.Context, params *backends.DeleteAllPa
 func (c *collection) Explain(ctx context.Context, params *backends.ExplainParams) (*backends.ExplainResult, error) {
 	p, err := c.r.DatabaseGetExisting(ctx, c.dbName)
 	if err != nil {
-		return nil, lazyerrors.Error(err)
+		return &backends.ExplainResult{
```
Comment (chilagrow) [inline internal/backends/postgresql/collection.go:264]: Isn't it simpler to handle `nil` in handler code? :thinking:

Comment (noisersup) [reply internal/backends/postgresql/collection.go:264]: That was my first approach, but after some time I realized we don't return nil in sqlite, so I've taken suggestion from there.

**Review by AlekSi — APPROVED** [review]

**Review by chilagrow — APPROVED** [review]

File: internal/backends/postgresql/collection.go (hunk 1 above)
Comment (chilagrow) [reply internal/backends/postgresql/collection.go:264]: All return non nil response.

**Pull request merged**